lib/Runtime/Debug/TTSnapObjects.cpp

```diff
@@ -2117,15 +2128,277 @@ namespace TTD
+
+            // TODO: BUGBUG - figure out how to determine what the prototype was.  Just use GetNull() for now
+            Js::RecyclableObject* prototype = ctx->GetLibrary()->GetNull();
+            //if (generatorInfo->generatorPrototype == 1) {
+            //    prototype = ctx->GetLibrary()->GetNull();
+            //}
+            //else if (generatorInfo->generatorPrototype == 2) {
+            //    prototype = ctx->GetLibrary()->CreateGeneratorConstructorPrototypeObject();
+            //}
+            //else {
+            //    //TTDAssert(false, "unexpected prototype found JavascriptGenerator");
+            //}
+
+            //Js::DynamicType* type = reinterpret_cast<Js::DynamicType*>((snpObject->SnapType->TypePtrId != TTD_INVALID_PTR_ID) ? inflator->LookupType(snpObject->SnapType->TypePtrId): nullptr);
+
+            return  ctx->GetLibrary()->CreateJavascriptGenerator_TTD(ctx, prototype, arguments, scriptFunction, static_cast<Js::JavascriptGenerator::GeneratorState>(generatorInfo->state));
+        }
+
+        void DoAddtlValueInstantiation_SnapGeneratorInfo(const SnapObject* snpObject, Js::RecyclableObject* obj, InflateMap* inflator) {
+            Js::ScriptContext* ctx = inflator->LookupScriptContext(snpObject->SnapType->ScriptContextLogId);
+            Js::JavascriptGenerator* generator = reinterpret_cast<Js::JavascriptGenerator*>(obj);
+
+            SnapGeneratorInfo* generatorInfo = SnapObjectGetAddtlInfoAs<SnapGeneratorInfo*, SnapObjectType::SnapGenerator>(snpObject);
+            Js::ScriptFunction* scriptFunction = reinterpret_cast<Js::ScriptFunction*>((generatorInfo->scriptFunction != TTD_INVALID_PTR_ID) ? inflator->LookupObject(generatorInfo->scriptFunction) : nullptr);
+            generator->SetScriptFunction(scriptFunction);
+
+            Js::FunctionBody* executeFunction = Js::JavascriptFunction::UnsafeFromVar(scriptFunction)->GetFunctionBody();
+
+            // call EnsureDynamicProfileInfo or we get asserts in some cases when createing the InterpreterStackFrame
+            executeFunction->EnsureDynamicProfileInfo();
+
+            bool doProfile = Js::InterpreterStackFrame::ShouldDoProfile(executeFunction);
+
+            Js::InterpreterStackFrame* frame = Js::InterpreterStackFrame::CreateInterpreterStackFrameForGenerator(scriptFunction, executeFunction, generator, doProfile);
+            TTDAssert(generator->GetFrame() == frame, "unexpected frame mis-match");
+
+            Js::Var* frameSlotArray = nullptr;
+            if (generatorInfo->frame_slotCount != 0) {
+                frameSlotArray = RecyclerNewArray(ctx->GetRecycler(), Field(Js::Var), generatorInfo->frame_slotCount);
+
+                for (Js::RegSlot i = 0; i < generatorInfo->frame_slotCount; i++) {
+                    if (generatorInfo->frame_slotArray[i] != nullptr) {
+                        frameSlotArray[i] = inflator->InflateTTDVar(generatorInfo->frame_slotArray[i]);
+                    }
+                    else {
```

Comment:
nit: braces should be on the following line (this applies to all the braces you've added/changed)

Reply:
should be fixed now.